conans/client/source.py

```diff
@@ -85,18 +85,19 @@ def remove_source(raise_error=True):
 
 
 def config_source_local(current_path, conan_file, output):
-    output.info('Configuring sources in %s' % current_path)
-    dirty = os.path.join(current_path, DIRTY_FILE)
-    if os.path.exists(dirty):
-        output.warn("Your previous source command failed")
+    with tools.chdir(current_path):
+        output.info('Configuring sources in %s' % current_path)
```

Comment:
I think the value of "dirty" in the local flow is not clear. Maybe remove it and simplify?

Reply:
Sorry for the confusion, it's removed in the other PR. It will have conflicts. So, merge first the other PR and then I'll update this.